src/CommunityToolkit.Maui/ImageSources/GravatarImageSource.shared.cs

```diff
@@ -0,0 +1,237 @@
+	const int defaultSize = 80;
+	static readonly HttpClient singletonHttpClient = new();
+	int gravatarSize;
+	CancellationTokenSource? tokenSource;
+
+	/// <summary>Initializes a new instance of the <see cref="GravatarImageSource"/> class.</summary>
+	public GravatarImageSource()
+	{
+		Stream = new Func<CancellationToken, Task<Stream>>(cancelationToken => DownloadStreamAsync(Uri, cancelationToken));
+		Uri = new Uri(defaultGravatarImageAddress);
+	}
+
+	/// <summary>Gets or sets a <see cref="TimeSpan"/> structure that indicates the length of time after which the image cache becomes invalid.</summary>
+	public TimeSpan CacheValidity
+	{
+		get => (TimeSpan)GetValue(CacheValidityProperty);
+		set => SetValue(CacheValidityProperty, value);
+	}
+
+	/// <summary>Gets or sets a Boolean value that indicates whether caching is enabled on this <see cref="GravatarImageSource"/> object.</summary>
+	public bool CachingEnabled
+	{
+		get => (bool)GetValue(CachingEnabledProperty);
+		set => SetValue(CachingEnabledProperty, value);
+	}
+
+	/// <summary>Gets or sets the email address.</summary>
+	public string? Email
+	{
+		get => (string)GetValue(EmailProperty);
+		set => SetValue(EmailProperty, value);
+	}
+
+	/// <summary>Gets or sets the default image.</summary>
+	public DefaultImage Image
+	{
+		get => (DefaultImage)GetValue(ImageProperty);
+		set => SetValue(ImageProperty, value);
+	}
+
+	/// <summary>Gets a value indicating whether the control email is empty.</summary>
+	public override bool IsEmpty => string.IsNullOrEmpty(Email);
+
+	/// <summary>Gets or sets the URI for the image to get.</summary>
+	[System.ComponentModel.TypeConverter(typeof(UriTypeConverter))]
+	public Uri Uri { get; private set; }
+
+	/// <summary>Gets or sets the control size property.</summary>
+	int GravatarSize
+	{
+		get => gravatarSize;
+		set => gravatarSize = Math.Clamp(value, 1, 2048);
+	}
+
+	/// <summary>Returns the Uri as a string.</summary>
+	/// <returns>String of the URI.</returns>
+	public override string ToString()
+	{
+		return $"Uri: {Uri}, Email:{Email}, Size:{GravatarSize}, Image: {DefaultGravatarName(Image)}, CacheValidity:{CacheValidity}, CachingEnabled:{CachingEnabled}";
+	}
+
+	/// <summary>On parent set.</summary>
+	protected override void OnParentSet()
+	{
+		object element = Parent;
+		base.OnParentSet();
+		if (element is not VisualElement parentElement || parentElement is null)
+		{
+			GravatarSize = defaultSize;
+			return;
+		}
+
+		var height = parentElement.Height >= 0 ? parentElement.Height : parentElement.HeightRequest;
+		if (height <= 0)
+		{
+			height = defaultSize;
+		}
+
+		var width = parentElement.Width >= 0 ? parentElement.Width : parentElement.WidthRequest;
+		if (width <= 0)
+		{
+			width = defaultSize;
+		}
+
+		var sizeFromParent = (int)Math.Max(width, height);
+		if (GravatarSize != sizeFromParent)
+		{
+			GravatarSize = sizeFromParent;
+			HandleNewUriRequested(Email, Image);
+		}
+	}
```

Comment:
@GeorgeLeithead is #564 failing because the `AvatarView` hasn't measured itself for the first time? If so could we potentially bind to the `WidthProperty` and `HeightProperty` of the parent and react to when it's dimensions do change?

We would want to able to limit the number of times we request a new image though so perhaps we only do it on the first positive above 0 change?

Reply:
I think it's a problem with `AvatarView` specifically, as it works with other controls that have the `ImageSource` property and have their appropriate `HeighrRequest` and/or `WidthRequest` set.  As such, I don't want to change `GavatarImageSource` to fix a problem specifically with `AvatarView`.  I think `AvatarView` is where the fix specifically for this, should be located.